code/game/turfs/open/floor/misc_floor.dm

```diff
@@ -190,3 +190,122 @@
 	base_icon_state = "tcomms"
 	icon = 'icons/turf/floors/misc.dmi'
 	color = null
+
+//ship turfs
+/turf/open/floor/ship
+	name = "Ship Plating"
+	desc = "Report a bug if you see this."
+
+/turf/open/floor/ship/dirt
+	gender = PLURAL
+	name = "dirt"
+	desc = "Upon closer examination, it's still dirt."
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "dirt"
+	footstep = FOOTSTEP_SAND
+	barefootstep = FOOTSTEP_SAND
+	clawfootstep = FOOTSTEP_SAND
+	heavyfootstep = FOOTSTEP_GENERIC_HEAVY
+	tiled_dirt = FALSE
+	var/ore_type = /obj/item/stack/ore/glass
+	var/turfverb = "dig up"
+	baseturfs = /turf/open/floor/plating
+
+/turf/open/floor/ship/dirt/attackby(obj/item/C, mob/user, params)
+	if((C.tool_behaviour == TOOL_SHOVEL) && params)
+		new ore_type(src, 2)
+		user.visible_message("<span class='notice'>[user] digs up [src].</span>", "<span class='notice'>You [turfverb] [src].</span>")
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, TRUE)
+		make_plating()
+	if(..())
+		return
+
+/turf/open/floor/ship/dirt/dark
+	icon_state = "greenerdirt"
+
+/turf/open/floor/grass/ship
```

Comment:
is this intended to be /turf/open/floor/ship/grass?

Reply:
No. Greener dirt is a subtype of jungle dirt.  This is how it is supposed to be.